tracim/tracim
# feat(backend): support ES indices for user and workspace

See #4147.

## Checkpoints



**For developers**

  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified

**For code reviewers**


**For testers**


**Testing method**

- launch third-party services with `docker-compose up -d` from backend directory
- edit `development.ini` to set:

```ini
search.engine = elasticsearch
search.elasticsearch.index_alias_prefix = tracim
```

- activate python virtual env
- check that `tracimcli search index-create` creates 3 indices prefixed by `tracim-(content|user|workspace)`. You can check by browsing to [http://localhost:9200/_stats](http://localhost:9200/_stats)
- check that `tracimcli search index-populate` finishes correctly
- - check that `tracimcli search index-upgrade-experimental` finishes correctly
- check that `tracimcli search index-drop` deletes the 3 indices prefixed by `tracim-(content|user|workspace)`.
